File: tests/human/rust/dataset_where_filter.rs
#[derive(Clone)]
struct Person { name: &'static str, age: i32 }

fn main() {
    let people = vec![
        Person { name: "Alice", age: 30 },
        Person { name: "Bob", age: 15 },
        Person { name: "Charlie", age: 65 },
        Person { name: "Diana", age: 45 },
    ];

    let adults: Vec<_> = people
        .into_iter()
        .filter(|p| p.age >= 18)
        .map(|p| (p.name, p.age, p.age >= 60))
        .collect();

    println!("--- Adults ---");
    for (name, age, is_senior) in adults {
        if is_senior {
            println!("{} is {}  (senior)", name, age);
        } else {
            println!("{} is {}", name, age);
        }
    }
}

File: archived/tests/transpiler/x/php/dataset_where_filter.php
<?php
$people = [["name" => "Alice", "age" => 30], ["name" => "Bob", "age" => 15], ["name" => "Charlie", "age" => 65], ["name" => "Diana", "age" => 45]];
$adults = [];
foreach ($people as $person) {
  if ($person["age"] >= 18) {
    $adults[] = ["name" => $person["name"], "age" => $person["age"], "is_senior" => $person["age"] >= 60];
  }
}

echo "--- Adults ---", PHP_EOL;
foreach ($adults as $person) {
  echo (is_float($person["name"]) ? json_encode($person["name"], 1344) : $person["name"]) . " " . "is" . " " . (is_float($person["age"]) ? json_encode($person["age"], 1344) : $person["age"]) . " " . ($person["is_senior"] ? " (senior)" : ""), PHP_EOL;
}
